test/storage/index_builder_test.cpp

```diff
@@ -0,0 +1,179 @@
+#include <string>
+#include <vector>
+#include "catalog/catalog.h"
+#include "catalog/catalog_sql_table.h"
+#include "storage/index/index_factory.h"
+#include "storage/index/index_manager.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier::storage::index {
+struct IndexBuilderTest : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+    index_manager_ = new IndexManager();
+    txn_ = txn_manager_->BeginTransaction();
+    catalog_ = new catalog::Catalog(txn_manager_, txn_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    TerrierTest::TearDown();
+    delete catalog_;  // delete catalog first
+    delete index_manager_;
+    delete txn_manager_;
+    delete txn_;
+  }
+
+  catalog::Catalog *catalog_;
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+
+  transaction::TransactionContext *txn_;
+  transaction::TransactionManager *txn_manager_;
+  IndexManager *index_manager_;
+};
+
+// Check the basic correctness of index populator
+// NOLINTNEXTLINE
```

Comment:
[Major] Now that `index_builder` has been changed to `index_manager`, this test, which tests `PopulateIndex()`, ought to be integrated into `index_manager_test`.

Reply:
We decide to keep the test for populating index. The reason is that if someone in the future changes the implementation of populating index, they can reuse our test to check the correctness of their implementation.